lib/Arch/PPC_C.cpp

```diff
@@ -0,0 +1,332 @@
+/*
+ * Copyright (c) 2022-present Trail of Bits, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <anvill/Declarations.h>
+#include <glog/logging.h>
+#include <remill/Arch/Arch.h>
+#include <remill/Arch/Name.h>
+
+#include "AllocationState.h"
+#include "Arch.h"
+
+namespace anvill {
+namespace {
+
+static const std::vector<RegisterConstraint> kParamRegConstraints = {
+    // GPRs
+    RegisterConstraint({VariantConstraint("R3", kTypeIntegral, kMaxBit32)}),
+    RegisterConstraint({VariantConstraint("R4", kTypeIntegral, kMaxBit32)}),
+    RegisterConstraint({VariantConstraint("R5", kTypeIntegral, kMaxBit32)}),
+    RegisterConstraint({VariantConstraint("R6", kTypeIntegral, kMaxBit32)}),
+    RegisterConstraint({VariantConstraint("R7", kTypeIntegral, kMaxBit32)}),
+    RegisterConstraint({VariantConstraint("R8", kTypeIntegral, kMaxBit32)}),
+    RegisterConstraint({VariantConstraint("R9", kTypeIntegral, kMaxBit32)}),
+    RegisterConstraint({VariantConstraint("R10", kTypeIntegral, kMaxBit32)}),
+    // FPRs
+    RegisterConstraint({VariantConstraint("F1", kTypeFloat, kMaxBit32)}),
+    RegisterConstraint({VariantConstraint("F2", kTypeFloat, kMaxBit32)}),
+    RegisterConstraint({VariantConstraint("F3", kTypeFloat, kMaxBit32)}),
+    RegisterConstraint({VariantConstraint("F4", kTypeFloat, kMaxBit32)}),
+    RegisterConstraint({VariantConstraint("F5", kTypeFloat, kMaxBit32)}),
+    RegisterConstraint({VariantConstraint("F6", kTypeFloat, kMaxBit32)}),
+    RegisterConstraint({VariantConstraint("F7", kTypeFloat, kMaxBit32)}),
+    RegisterConstraint({VariantConstraint("F8", kTypeFloat, kMaxBit32)}),
+    RegisterConstraint({VariantConstraint("F9", kTypeFloat, kMaxBit32)}),
+    RegisterConstraint({VariantConstraint("F10", kTypeFloat, kMaxBit32)}),
+    RegisterConstraint({VariantConstraint("F11", kTypeFloat, kMaxBit32)}),
+    RegisterConstraint({VariantConstraint("F12", kTypeFloat, kMaxBit32)}),
+    RegisterConstraint({VariantConstraint("F13", kTypeFloat, kMaxBit32)}),
+};
+
+static const std::vector<RegisterConstraint> kReturnRegConstraints = {
+    // GPRs
+    RegisterConstraint({VariantConstraint("R3", kTypeIntegral, kMaxBit32)}),
+    RegisterConstraint({VariantConstraint("R4", kTypeIntegral, kMaxBit32)}),
+    // FPRs
+    RegisterConstraint({VariantConstraint("F1", kTypeFloat, kMaxBit32)}),
+    RegisterConstraint({VariantConstraint("F2", kTypeFloat, kMaxBit32)}),
```

Comment:
F1 is only used for return value

Reply:
That doc you linked seems to suggest that it's also used for passing parameters.

Reply:
ah yeah, misworded
i meant to say that F2-F4 aren't used as return registers
looks like you fixed this

Reply:
Oh right, understood.